components/global_search/global-search.react.js

```diff
@@ -0,0 +1,146 @@
+import React, {Component} from 'react';
+import './style.scss';
+import SearchOption from "./child_components/search_option/SearchOption";
+
+class GlobalSearch extends Component {
+
+    constructor(props) {
+        super(props);
+
+        this.handle_search_selection = this.handle_search_selection.bind(this);
```

Comment:
@AshTNA , you could possibly cut your code down a lot, if you use `arrow functions`.
This way you don't need to bind your methods.